bundles/org.eclipse.mita.program/src/org/eclipse/mita/program/generator/transformation/PrepareArrayRuntimeChecksStage.xtend

```diff
@@ -13,37 +13,73 @@
 
 package org.eclipse.mita.program.generator.transformation
 
+import com.google.inject.Inject
+import org.eclipse.mita.base.expressions.ElementReferenceExpression
+import org.eclipse.mita.base.expressions.PrimitiveValueExpression
 import org.eclipse.mita.program.AbstractLoopStatement
 import org.eclipse.mita.program.ArrayAccessExpression
 import org.eclipse.mita.program.DoWhileStatement
 import org.eclipse.mita.program.ForStatement
+import org.eclipse.mita.program.Program
 import org.eclipse.mita.program.ProgramBlock
-import org.eclipse.mita.program.WhileStatement
-import org.eclipse.xtext.EcoreUtil2
 import org.eclipse.mita.program.ProgramFactory
+import org.eclipse.mita.program.ValueRange
+import org.eclipse.mita.program.WhileStatement
 import org.eclipse.mita.program.inferrer.ElementSizeInferrer
-import com.google.inject.Inject
 import org.eclipse.mita.program.inferrer.StaticValueInferrer
 import org.eclipse.mita.program.inferrer.ValidElementSizeInferenceResult
+import org.eclipse.xtext.EcoreUtil2
 
 class PrepareArrayRuntimeChecksStage extends AbstractTransformationStage {
 	
 	@Inject
-	protected ElementSizeInferrer sizeInferrer
+	protected ElementSizeInferrer sizeInferrer;
+	
+	@Inject
+	protected UnravelLiteralArrayReturnStage unravelExpression;
```

Comment:
AFAIK this is the first time we're introducing explicit dependencies/coupling between pipeline stages

Reply:
It's more like reusing code though, the stage doesn't really run.